gammapy/spectrum/utils.py

```diff
@@ -13,19 +16,254 @@
+        Parameters
+        ----------
+        bin_centers : `~astropy.units.Quantity`
+        Energy bin centers
+
+        Returns
+        -------
+        Energy Binning
+        """
+
+        # TODO: test for equal log spacing
+        bin_edges = find_log_edges(bin_centers)
+
+        return EnergyBinning(bin_edges, bin_centers)
+
+    @staticmethod
+    def equal_log_spacing(emin, emax, nbins):
+        """Create EnergyBinning with equal log-spacing by giving the outer bin edges
+
+        Parameters
+        ----------
+        emin : `~astropy.units.Quantity`
+        Lower bound of lowest energy bin
+        emax : `~astropy.units.Quantity`
+        Upper bound of highest energy bin
+        bins : int
+        Number of bins
+
+        Returns
+        -------
+        Energy Binning
+        """
+
+        if not isinstance(emin, Quantity) or not isinstance(emax, Quantity):
+            raise ValueError("Energies must be Quantity objects.")
+
+        emin = emin.to('TeV')
+        emax = emax.to('TeV')
+
+        x_min, x_max = np.log10([emin.value, emax.value])
+        bin_edges = np.logspace(x_min, x_max, nbins + 1)
+
+        bin_edges = Quantity(bin_edges, emin.unit)
+
+        return EnergyBinning.from_edges(bin_edges, 'log')
+
+    @staticmethod
+    def from_fits(hdulist):
+        """Read energy axis from hdulist
+
+        Support all formats here
+        """
+
+        try:
+            bin_centers = Quantity(hdulist['ENERGIES'].data['Energy'], 'MeV')
+            return EnergyBinning.from_centers(bin_centers)
+        except KeyError:
+            pass
+
+        try:
+            energy_lo = hdulist['SPECRESP'].data['ENERG_LO']
+            energy_hi = hdulist['SPECRESP'].data['ENERG_HI']
+            bin_edges = Quantity(energy_lo + energy_hi[-1],
+                                 hdulist['SPECRESP'].header['TUNIT1'])
+            return EnergyBinning.from_edges(bin_edges, centers='log')
+        except KeyError:
+            pass
+
+        raise ValueError("The hdulist does not contain a valid energy extension")
+
+    def to_fits(self, name, **kwargs):
+        """Write energy axis to hdulist
+        """
+
+        if name == 'ENERGIES':
+            #this sould go to a separate function in the end I guess
+            #TODO write header, and so on
+            col1 = fits.Column(name='Energy', format='D', array=self.bin_centers)
+            cols = fits.ColDefs([col1])
+            return fits.BinTableHDU.from_columns(cols)
+
+    @property
+    def bin_centers(self):
+        """
+        Centers of the energy bins
+        """
+
+        if self._bin_centers is not None:
+            return self._bin_centers
+        else:
+            raise ValueError("Bin centers not defined for this energy axis")
```

Comment:
This is a bad API, I think (too complicated): properties that are sometimes there and sometimes not.